### PR TITLE
Js task 8

### PR DESCRIPTION
**Pull request by dudzon**
https://mogopopup.netlify.com/

### CODE REVIEW

File: css_tasks/task11/js/popup.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import {elements} from './elements.js';
+
+export function popup() {
+  elements.then((data) => {
+    const {gallery, popup, body, overlay} = data;
```
Comment (tkachukivan) [inline css_tasks/task11/js/popup.js:5]: you can use destructuring in function variable ie ({gallery, popup, body, overlay}) => {}

File: css_tasks/task11/js/popup.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    gallery.addEventListener('click', (e) => showPopup(e.target));
+    overlay.addEventListener('click', (e) => removePopup());
+    const showPopup = (node) => {
+      // if popup is displayed, do nothing
+
+      if (popup.firstElementChild) {
```
Comment (tkachukivan) [inline css_tasks/task11/js/popup.js:11]: why not check by class?

File: css_tasks/task11/js/popup.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return;
+      }
+      createActiveImage(node);
+
+      //  additional styling to properly display popup
+      popup.style.display = 'flex';
```
Comment (tkachukivan) [inline css_tasks/task11/js/popup.js:17]: classes with prefixes like active and so on would be much better

File: css_tasks/task11/js/popup.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      body.classList.add('popup-body');
+      overlay.style.display = 'block';
+    };
+
+    //  pick active image
+    const setActiveImage = (node) => {
+      const image = node.closest('.gallery__element').firstElementChild
```
Comment (tkachukivan) [inline css_tasks/task11/js/popup.js:24]: better to avoid traveling dom like this

File: css_tasks/task11/js/popup.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    };
+
+    // create active image and populate popup
+    // I played with data attributes but in this
+    // case, it is not really needed right now
+
+    const createActiveImage = (node) => {
+      const active = setActiveImage(node);
+      const currentSrc = active.src;
+      const image = document.createElement('IMG');
```
Comment (tkachukivan) [inline css_tasks/task11/js/popup.js:36]: can be lower cased, there is also Image constructor

File: css_tasks/task11/js/popup.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      image.src = currentSrc;
+      image.setAttribute('data-image-active', 'active');
+      popup.appendChild(image);
+    };
+
+    //  remove popup - remove image in popup from DOM
+    //  styling to hide overlay
+
+    const removePopup = () => {
+      popup.style.display = 'none';
```
Comment (tkachukivan) [inline css_tasks/task11/js/popup.js:46]: classes?

File: css_tasks/task11/js/popup.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      createActiveImage(node);
+
+      //  additional styling to properly display popup
+      popup.style.display = 'flex';
+      body.classList.add('popup-body');
+      overlay.style.display = 'block';
+    };
+
+    //  pick active image
+    const setActiveImage = (node) => {
```
Comment (tsvetkovv) [inline css_tasks/task11/js/popup.js:23]: `getActiveImage`? `set..` does not assume to return smth

**Pull request merged**